ManageWikiSettings.php

```diff
@@ -872,6 +872,32 @@
 		'help' => 'Number of seconds before a RottenLinks request returns no response. Min: 5. Max: 120.',
 		'requires' => [],
 	],
+	'wgRelatedArticlesCardLimit' => [
+		'name' => 'Related Articles Card Limit',
+		'from' => 'relatedarticles',
+		'type' => 'integer',
+		'minint' => 1,
+		'maxint' => 20,
+		'overridedefault' => 3,
+		'section' => 'links',
+		'help' => 'Maximum number of articles that should be shown in RelatedArticles widget. This limit is derived from limits in TextExtracts and PageImages extensions.',
+		'requires' => [],
+	],
+	'wgRelatedArticlesDescriptionSource' => [
+		'name' => 'Related Articles Description Source',
+		'from' => 'relatedarticles',
+		'type' => 'list',
+		'options' => [
```

Comment:
All 3 of these require a different extension (if to be changed at all) this should maybe be done similarly to how the theme config is done at the very end of this file (preferably don't do it like that though) or add a requires to require one of the other extensions for this config also. If you need an exact example of what requires could be I can get something probably. Also these options are quite long and might not be ideal display names.

Reply:
I didn't understand the latter and didn't guess why shouldn't do the former. If it is by the split code, how about the below use of `array_filter()`?

```php
[
  'options' => [
      'None' => false,
    ] + array_filter( [
      'From the central wikibase repository.' =>
        ( $wi->config->get( 'wmgUseWikibaseRepository', $wi->dbname ) || $wi->config->get( 'wmgUseShortDescription', $wi->dbname ) ) ? 'wikidata' : false,
      'Fetched from the prop=extracts api.' => $wi->config->get( 'wmgUseWikibaseRepository', $wi->dbname ) ? 'textextracts' : false,
      'From the description page property.' =>
        ( $wi->config->get( 'wmgUseWikiSeo', $wi->dbname ) || $wi->config->get( 'wmgUseDescription2', $wi->dbname ) ) ? 'pagedescription' : false,
    ] ),
];
```

And though it is another topic, should wgCosmosSearchDescriptionSource also change?
https://github.com/miraheze/mw-config/blob/fb3b71f1f1b156a08222f162da489cc4eb98a315/ManageWikiSettings.php#L2718-L2722

Reply:
I would rather just do:
```php
'requires' => [
	'extensions' => [
		'relatedarticles',
		[
			'description2',
			'shortdescription',
			'textextracts',
			'wikibaseclient',
			'wikiseo',
		],
	],
],
```
That's a cleaner way. It basically makes it require RelatedArticles, as well as **OR** one of the others. More can be added if necessary. This also depends on if this support is properly implemented so it will work with ManageWikiSettings, as it should with extensions, but I'm not 100% sure. I know this way doesn't limit per option, but in my opinion is OK, as it does at least condition it to require one of them that'll actually do something. But maybe we should just not do anything here?

Reply:
> This also depends on if this support is properly implemented so it will work with ManageWikiSettings, as it should with extensions, but I'm not 100% sure.

Sorry, I don't understand what this sentence means.

> I know this way doesn't limit per option, but in my opinion is OK, as it does at least condition it to require one of them that'll actually do something.

If you are OK, I would be OK too. I just want to know if the condition of the option is not satisfied, no exception would be thrown.

> But maybe we should just not do anything here?

https://github.com/miraheze/mw-config/pull/4474#issuecomment-1046346371 is my indirect answer to this.

And, thank you for reviewing :smiley:

Reply:
> Sorry, I don't understand what this sentence means.

I'm just not sure if this AND OR logic is supported by ManageWikiSettings. I know it is with ManageWikiExtensions (or should be) but I'm not sure beyond that. But I'd rather do this way then the filter, just because it's less messy.

Reply:
Oh, I agree it is less messy. And I'm also not sure the logic works, I just found VEForAll, which "requires: Extensions - ( commentstreams OR pageforms ), visualeditor",

https://github.com/miraheze/mw-config/blob/fb3b71f1f1b156a08222f162da489cc4eb98a315/ManageWikiExtensions.php#L3201-L3213

but it is defined via `$wgManageWikiExtensions` not `$wgManageWikiSettings`.

Reply:
I believe I did expand the AND OR logic to ManageWikiSettings awhile ago. It should work here also.

Reply:
Yes, I did: https://github.com/miraheze/ManageWiki/blob/772b596f75c06016d7c01e361a48b9c1d2e99af5/includes/formFactory/ManageWikiFormFactoryBuilder.php#L355